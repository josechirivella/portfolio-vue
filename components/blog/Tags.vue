<template>
  <div class="px-6 pt-4 pb-2">
    <template v-for="(tag, index) in props.tags" :key="index">
      <span class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2">#{{tag}}</span>
    </template>
  </div>
</template>

<script lang="ts" setup>
const props = defineProps({
  tags: { type: Array<string>, required: true }
});

</script>
