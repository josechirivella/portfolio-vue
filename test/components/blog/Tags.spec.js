import { beforeAll, describe, expect, test } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Tags from '~/app/components/blog/Tags.vue';

describe('Tags', () => {
  let wrapper;
  const mockTags = ['vue', 'javascript', 'testing'];

  beforeAll(() => {
    wrapper = shallowMount(Tags, {
      props: {
        tags: mockTags,
      },
    });
  });

  test('should render component', () => {
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  describe('Tags rendering', () => {
    test('should render the correct number of tags', () => {
      const tagElements = wrapper.findAll('.inline-block');
      expect(tagElements.length).toBe(mockTags.length);
    });

    test('should render the correct tag text', () => {
      const tagElements = wrapper.findAll('.inline-block');

      mockTags.forEach((tag, index) => {
        expect(tagElements[index].text()).toBe(`#${tag}`);
      });
    });
  });

  describe('Props', () => {
    test('should receive tags prop correctly', () => {
      expect(wrapper.props('tags')).toEqual(mockTags);
    });

    test('should update when tags prop changes', async () => {
      const newTags = ['nuxt', 'typescript'];

      await wrapper.setProps({
        tags: newTags,
      });

      const tagElements = wrapper.findAll('.inline-block');
      expect(tagElements.length).toBe(newTags.length);

      newTags.forEach((tag, index) => {
        expect(tagElements[index].text()).toBe(`#${tag}`);
      });
    });
  });
});
