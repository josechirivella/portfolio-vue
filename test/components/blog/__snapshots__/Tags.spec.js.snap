// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tags > should render component 1`] = `
<div
  class="px-6 pt-4 pb-2"
>
  
  <span
    class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
  >
    #vue
  </span>
  <span
    class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
  >
    #javascript
  </span>
  <span
    class="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
  >
    #testing
  </span>
  
</div>
`;
